libfwupdplugin/fu-udev-device.c

```diff
@@ -1252,6 +1217,85 @@ fu_udev_device_ioctl(FuUdevDevice *self,
 #endif
 }
 
+/**
+ * fu_udev_device_ioctl:
+ * @self: a #FuUdevDevice
+ * @request: request number
+ * @buf: a buffer to use, which *must* be large enough for the request
+ * @bufsz: the size of @buf
+ * @rc: (out) (nullable): the raw return value from the ioctl
+ * @timeout: timeout in ms for the retry action, see %FU_UDEV_DEVICE_IOCTL_FLAG_RETRY
+ * @flags: some #FuUdevDeviceIoctlFlags, e.g. %FU_UDEV_DEVICE_IOCTL_FLAG_RETRY
+ * @error: (nullable): optional return location for an error
+ *
+ * Control a device using a low-level request.
+ *
+ * NOTE: In version 2.0.0 the @bufsz parameter was added -- which isn't required to perform the
+ * ioctl, but *is* required to accurately track and emulate the device buffer.
+ *
+ * Returns: %TRUE for success
+ *
+ * Since: 2.0.0
```

Comment:
2.0.2 since this is changing no?

Reply:
I removed the gtk-doc as it's now private.